lib/src/Uncertainty/Distribution/EmpiricalBernsteinCopula.cxx

```diff
@@ -335,6 +344,210 @@ Scalar EmpiricalBernsteinCopula::computeProbability(const Interval & interval) c
 }
 
 
+/* Compute the PDF of Xi | X1, ..., Xi-1. x = Xi, y = (X1,...,Xi-1)
+   The empirical Bernstein copula is a mixture of products of Beta distributions:
+   c_m(u_1,...,u_d)=1/n\sum_{i=1}^n\prod_{j=1}^d \beta_{r_j^i,s_j^i}(u_j)
+   where r_j^i=\lceil mU_j^i\rceil, s_j^i=m+1-r_j^i
+   and \beta_{r,s}(x)=x^{r-1}(1-x)^{s-1}/B(r,s)
+   Here \log(r_j^i) is stored into logFactors_(i,j)=LF(i,j),
+   B(r_j^i,s_j^i) is stored into logBetaMarginalFactors_(i,j)=LBMF(i,j) and
+   \sum_{j=1}^d\log(Beta(r_j^i,s_j^i))=\sum_{j=1}^d LBMF(i,j) is stored into logBetaFactors_(i)=LBF(i)
+   so c_m(u_1,...,u_d)=1/n\sum_{i=1}^n\exp(LBF(i)+\sum_{j=1}^d [(LF(i,j)-1)\log(u_j)+(m-LF(i,j))\log(1-u_j)])
+   
+   c_m(u_1)=1/n\sum_{i=1}^n\exp(LBMF(i,0)+[(LF(i,j)-1)\log(u_j)+(m-LF(i,j))\log(1-u_j)])
+   c_m(u_k|u_1,...,u_{k-1})=c_m(u_1,...,u_k)/c_m(u_1,...,u_{k-1})
+                           =\sum_{i=1}^n\exp(LBF(i)+\sum_{j=1}^k [(LF(i,j)-1)\log(u_j)+(m-LF(i,j))\log(1-u_j)])/\sum_{i=1}^n\exp(LBF(i)+\sum_{j=1}^{k-1}[(LF(i,j)-1)\log(u_j)+(m-LF(i,j))\log(1-u_j)])
+
+*/
+Scalar EmpiricalBernsteinCopula::computeConditionalPDF(const Scalar x,
+    const Point & y) const
+{
+  const UnsignedInteger conditioningDimension = y.getDimension();
+  if (conditioningDimension >= getDimension()) throw InvalidArgumentException(HERE) << "Error: cannot compute a conditional PDF with a conditioning point of dimension greater or equal to the distribution dimension.";
+  if (x <= 0.0 || x >= 1.0) return 0.0;
+  for (UnsignedInteger i = 0; i < y.getDimension(); ++i)
+    if (y[i] <= 0.0 || y[i] >= 1.0) return 0.0;
+  const UnsignedInteger size = copulaSample_.getSize();
+  // Special case for no conditioning or independent copula
+  if ((conditioningDimension == 0) || (hasIndependentCopula()))
+    {
+      if (isCopula()) return 1.0;
```

Comment:
I do not understand what it would mean for an `EmpiricalBernsteinCopula` not to be a copula.

Reply:
This commit should never have landed here. I messed things yesterday. Concerning your question, a better place would be to discuss it on PR#1646. What is called a Bernstein copula is a Bernstein approximation of a copula function over a regular grid. When applied to the empirical copula (which is NOT a copula despite its name: the marginal distributions are not uniform over [0,1]), one get an absolutely continuous function which is an actual copula if and only if the marginal grid size is a proper divider of the sample size. So yes, an empirical Bernstein copula can be something else than a copula, and it is useful to have it e.g. for estimation purpose such as the MIIC algorithm in otagrum.

Reply:
All right thanks for your explanation.